### DISCUSSION TITLE
Self hosting Ente Server is throwing an error

### DISCUSSION THREAD

**Discussion by Sandeepbreddy** [ente-io/ente#4346] [General]
Self hosting Ente Server using ENTE docker image seeing below error, all the other containers are running fine.

Error 
museum-1           | INFO[0000]main.go:95 main Booting up local server with commit #b0674cd5f37cebda0f446db7ce1e87a3e6d8b9fc 
museum-1           | INFO[0000]main.go:791 setupDatabase Setting up db                                
museum-1           | INFO[0000]main.go:798 setupDatabase Connected to DB                              
museum-1           | panic: dial tcp [::1]:5432: connect: connection refused
museum-1           | 
museum-1           | goroutine 1 [running]:
museum-1           | main.setupDatabase()
museum-1           | 	/etc/ente/cmd/museum/main.go:801 +0x36b
museum-1           | main.main()

passing below details in credentials file

db:
    host: postgres
    port: 5432 
    name: ente_db
    user: test
    password: test

tried to connect to [::1]:5432 with in docker container which is connecting to my db.

Is there a way Ente/Server can use the host IPv4 address instead IPv6. Both Museum, Postgres are on same network.

**Comment by mngshm** [comment]
Port `:::5432` seems to be occupied somehow, can you do `sudo lsof -i :5432` and kill the process with the PID. Additionally also do 
`sudo ps aux | grep [p]ostgres | awk '{ print $2}'` and kill those processes as well. And do make sure the check the permissions in your postgres versions `pg_hba.conf`. The reason also could be that postgres is not accepting ipV6 (`::1`)connections

**Reply by Sandeepbreddy** [reply]
Tried the steps you mentioned above, not seeing any process with 5432

![image](https://github.com/user-attachments/assets/dadc4fcb-eb2e-43da-ae45-abc70a8a88c1)


![image](https://github.com/user-attachments/assets/dc4a266e-9216-441e-81ef-b8831e0f6ca1)

-> docker ps 

![image](https://github.com/user-attachments/assets/3e217aa0-b7a2-4648-bee6-f92d9ef6527d)

-> postgresql logs from compose


![image](https://github.com/user-attachments/assets/32a90e0a-36c3-4807-ac6f-6bf988a017a9)

**Comment by Sandeepbreddy** [comment]
Resolved the issue, my bad there are no file permission. So it refused

**Reply by Emilv2** [reply]
I'm having the same issue, what did you do to fix it?

**Reply by etuckeriv** [reply]
@Sandeepbreddy Im also having the same issue. What did you do to resolve?

**Reply by Sandeepbreddy** [reply]
Sorry for my late response. The only way I resolved this issue is,
      - /home/ubuntu/docker/ente/custom-logs:/var/logs
      - /home/ubuntu/docker/ente/museum.yaml:/museum.yaml:ro
      - /home/ubuntu/docker/ente/scripts/compose/credentials.yaml:/credentials.yaml:ro
      - /home/ubuntu/docker/ente/data:/data:ro
I gave the volumes part but the folders doesn't have permissions so the IPv6 loopback test is failing

So I manually given the permissions.

And I verified if the port are open, as am using iptables so manually added the ports

**Reply by mngshm** [reply]
@Sandeepbreddy once the discussion ends and you share commands or so. Please mark this discussion as answered to your reply :).

**Reply by etuckeriv** [reply]
@Sandeepbreddy no worries, thanks for the reply! My issue ended up being that I mis-configured the volume mounts and the museum container couldn't find my Postgres credentials.
